src/promptflow/promptflow/_sdk/_submitter/utils.py

```diff
@@ -111,26 +111,29 @@ def overwrite_connections(flow_dag: dict, connections: dict, working_dir: PathLi
             raise InvalidFlowError(f"Invalid connection overwrite format: {connection_dict}, only dict is supported.")
         node = node_name_2_node[node_name]
         executable_node = executable_flow.get_node(node_name=node_name)
+        unsupported_keys = connection_dict.keys() - SUPPORTED_CONNECTION_FIELDS
```

Comment:
I assume the connection, mode, deployment_name paramter will be dynamic for custom tools?

Reply:
Offline synced, will sync with cloud behavior in this PR.

Reply:
Added to do since we'll change cloud's behavior, will support enabled by then.